addon/components/dialog.ts

```diff
@@ -52,6 +52,26 @@ export default class DialogComponent extends Component<Args> {
     }
   );
 
+  lockWindowScroll = modifier(() => {
+    // Opt-out of some other dialog already locked scrolling
+    if (this.dialogStackProvider.dialogIsOpen) {
```

Comment:
We added this `dialogIsOpen` property to the service because there really wasn't an API for querying whether a dialog is open generically. The `hasOpenChild` method isn't really expressing what we want here; we want to know if this modal, being opened right now, is the "top most" modal. If it is, we want to run the scroll lock. If not, we will rely on the "top most" modal already having applied the lock, and trust they will clean it up as well.